pkgs/by-name/gl/gltron/package.nix

```diff
@@ -39,6 +39,12 @@ stdenv.mkDerivation rec {
   # The build fails, unless we disable the default -Wall -Werror
   configureFlags = [ "--disable-warn" ];
 
+  makeFlags = [ "AR=${stdenv.cc.targetPrefix}ar" ];
+
+  nativeBuildInputs = [
+    SDL # sdl-config
```

Comment:
despite running on the build machine, if it's `sdl-config` you want, you'll want the one from cross-compiled `SDL.dev` output. otherwise, `sdl-config --libs` will be giving info about how to link against the build SDL instead of the host SDL.

try e.g.:
```sh
$ nix-build -A pkgsCross.aarch64-multiplatform.SDL.dev
$ ./result/bin/sdl-config --libs
-L/nix/store/a4xxnx2f179fxgwhh0l08pz7q2106ikp-SDL-aarch64-unknown-linux-gnu-1.2.15/lib ...
```
v.s. the same for `nix-build -A SDL.dev` (which gives a -L flag, and others, which point to the build SDL).

if it works here too i'd say go with the same `SDL_CONFIG` env fix you did over in https://github.com/NixOS/nixpkgs/pull/367712

Reply:
I'm fairly certain I checked that this change works (after applying the other PR), but I will check again.

Reply:
it might still work, same way having a native dependency in `buildInputs` sometimes still works, or how a header-only library placed in `nativeBuildInputs` sometimes still works. but as in those cases, it's fragile: it only works so long as the `sdl-config --cflags` never outputs any target-specific flags, for example -- a thing we can't generally guarantee.